proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,190 @@
+
+
+### Constants
+
+
+| Global Constants                |         |                                                                                                            |
+|:-------------------------------:|:-------:|------------------------------------------------------------------------------------------------------------|
+| **Name**                        |**Value**|**Description**                                                                                             |
+| `MODULE_ID_LENGTH_BYTES `       | 4       | The length of module IDs.                                                                                  |
+| `COMMAND_ID_LENGTH_BYTES`       | 2       | The length of command IDs.                                                                                 |
+| `PUBLIC_KEY_LENGTH_BYTES `      | 32      | The length of public keys.                                                                                 |
+| `SIGNATURE_LENGTH_BYTES`        | 64      | The length of signatures.                                                                                  |
+| **Configurable Constants**      |         |                                                                                                            |
+| **Name**                        |**Mainchain Value**        |**Description**                                                                           |
+| `MAX_PARAMS_SIZE_BYTES`         | 14 KiB (14*1024 bytes)    |   The maximum allowed length of the transaction parameters.                             |
+
+
+### JSON Schema
+
+Transactions are serialized using `transactionSchema` given below.
+
+
+```java
+transactionSchema = {
+    "type": "object",
+    "required": [
+        "moduleID",
+        "commandID",
+        "nonce",
+        "fee",
+        "senderPublicKey",
+        "params",
+        "signatures"
+    ],
+    "properties": {
+        "moduleID": {
+            "dataType": "bytes",
+            "length": MODULE_ID_LENGTH_BYTES,
+            "fieldNumber": 1
+        },
+        "commandID": {
+            "dataType": "bytes",
+            "length": COMMAND_ID_LENGTH_BYTES,
+            "fieldNumber": 2
+        },
+        "nonce": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        },
+        "fee": {
+            "dataType": "uint64",
+            "fieldNumber": 4
+        },
+        "senderPublicKey": {
+            "dataType": "bytes",
+            "length": PUBLIC_KEY_LENGTH_BYTES,
+            "fieldNumber": 5
+        },
+        "params": {
+            "dataType": "bytes",
+            "fieldNumber": 6
+        },
+        "signatures": {
+            "dataType": "array",
+            "items": {
+                "dataType": "bytes",
+                "length": SIGNATURE_LENGTH_BYTES 
+            },
+            "fieldNumber": 7
+        }
+    }
+}
+```
+
+
+### Validation
+
+For a transaction `trs` to be valid, it must satisfy the following:
```

Comment:
Here the context of where this would be checked is missing.

Reply:
Do we actually have to decide ourselves where this is checked? I find this to be an implementation detail. Personally, I think this check should not be part of the application layer but rather of the block header processing (consensus). For instance, we could perform the check as part of the `transactionRoot` calculation (since we have to serialize transactions there anyway).

Reply:
Can be resolved from my side as it is now clearly defined.

Reply:
cbr